WalletWasabi/Blockchain/Transactions/TransactionFactoryParameters.cs

```diff
@@ -0,0 +1,22 @@
+using NBitcoin;
+using System.Collections.Generic;
+
+namespace WalletWasabi.Blockchain.Transactions;
+
+public record TransactionFactoryParameters
+{
+	public TransactionFactoryParameters(Func<FeeRate> feeRateFetcher, bool allowUnconfirmed = false, bool allowDoubleSpend = false, IEnumerable<OutPoint>? allowedInputs = null, bool tryToSign = true)
+	{
+		FeeRateFetcher = feeRateFetcher;
+		AllowUnconfirmed = allowUnconfirmed;
+		AllowDoubleSpend = allowDoubleSpend;
+		AllowedInputs = allowedInputs;
+		TryToSign = tryToSign;
+	}
+
+	public Func<FeeRate> FeeRateFetcher { get; }
+	public bool AllowUnconfirmed { get; }
+	public bool AllowDoubleSpend { get; }
+	public IEnumerable<OutPoint>? AllowedInputs { get; }
+	public bool TryToSign { get; }
+}
```

Comment:
OK, this become a `record` but it can be further simplified.

```suggestion
public record TransactionFactoryParameters(Func<FeeRate> FeeRateFetcher, bool AllowUnconfirmed = false, bool AllowDoubleSpend = false, IEnumerable<OutPoint>? AllowedInputs = null, bool TryToSign = true);
```